app/models/conversion_host.rb

```diff
@@ -158,14 +158,91 @@ def ipaddress(family = 'ipv4')
+      ssu.put_file("/tmp/#{task_id}-input.json", conversion_options.to_json)
+      ssu.shell_exec("mv /tmp/#{task_id}-input.json /var/lib/uci/#{task_id}/input.json", nil, nil, nil)
+    end
+  rescue MiqException::MiqInvalidCredentialsError, MiqException::MiqSshUtilHostKeyMismatch => err
+    raise "Failed to connect and prepare conversion for task '#{task_id}' with [#{err.class}: #{err}]"
+  rescue JSON::GeneratorError => err
+    raise "Could not generate JSON for task '#{task_id}' from options '#{filtered_options}' with [#{err.class}: #{err}]"
+  rescue => err
+    raise "Preparation of conversion for task '#{task_id}' failed on '#{resource.name}' with [#{err.class}: #{err}]"
+  end
+
+  # Checks that LUKS keys vault exists and is valid JSON
+  # We don't care about the file content, as virt-v2v-wrapper will check it later
+  #
+  # @return [Boolean] true if the file can be retrieved and parsed, false otherwise
+  #
+  # @raise [MiqException::MiqInvalidCredentialsError] if conversion host credentials are invalid
+  # @raise [MiqException::MiqSshUtilHostKeyMismatch] if conversion host key has changed
+  # @raise [JSON::ParserError] if file cannot be parsed as JSON
+  def luks_keys_vault_valid?
+    luks_keys_vault_json = connect_ssh { |ssu| ssu.get_file("/root/.v2v_luks_keys_vault.json", nil) }
+    JSON.parse(luks_keys_vault_json)
+    true
+  rescue MiqException::MiqInvalidCredentialsError, MiqException::MiqSshUtilHostKeyMismatch => err
+    raise "Failed to connect and retrieve LUKS keys vault from file '/root/.v2v_luks_keys_vault.json' with [#{err.class}: #{err}]"
+  rescue JSON::ParserError
+    raise "Could not parse conversion state data from file '/root/.v2v_luks_keys_vault.json': #{json_state}"
+  rescue
+    false
+  end
+
+  # Build the podman command to execute conversion
+  #
+  # @param [Integer] id of the task that conversion applies to
+  #
+  # @return [String] podman command to be executed on conversion host
+  def build_podman_command(task_id, conversion_options)
+    uci_settings = Settings.transformation.uci.container
+    uci_image = uci_settings.image
+    uci_image = "#{uci_settings.registry}/#{image}" if uci_settings.registry.present?
+
+    command = "/usr/bin/podman run --detach --privileged"
+    command += " --name conversion-#{task_id}"
+    command += " --network host"
+    command += " --volume /dev:/dev"
+    command += " --volume /etc/pki/ca-trust:/etc/pki/ca-trust"
+    command += " --volume /var/tmp:/var/tmp"
+    command += " --volume /var/lib/uci/#{task_id}:/var/lib/uci"
+    command += " --volume /root/.ssh/id_rsa:/var/lib/uci/ssh_private_key" if conversion_options[:transport_method] == 'ssh'
+    command += " --volume /root/.v2v_luks_keys_vault.json:/var/lib/uci/luks_keys_vault.json" if luks_keys_vault_valid?
+    command += " --volume /var/log/uci/#{task_id}:/var/log/uci"
+    command += " --volume /opt/vmware-vix-disklib-distrib:/opt/vmware-vix-disklib-distrib"
+    command + " #{uci_image}"
```

Comment:
Since this is coming from settings it is technically user input and should be better escaped IMO